core/src/main/scala/kafka/server/ControllerServer.scala

```diff
@@ -160,11 +160,24 @@ class ControllerServer(
       alterConfigPolicy = Option(config.
         getConfiguredInstance(AlterConfigPolicyClassNameProp, classOf[AlterConfigPolicy]))
 
+      if (config.interBrokerProtocolVersionString != Defaults.InterBrokerProtocolVersion) {
+        // TODO can we have a stronger check here?
```

Comment:
I think this check should be in KafkaConfig. It would also be stronger to look at `KafkaConfig#originals` to verify that that we weren't trying to set it to anything (i.e. ibp doesn't appear as a key in that map)